internal/database/prune.go

```diff
@@ -5,14 +5,24 @@ import (
 	"time"
 )
 
-func DownsampleOverTime(interval time.Duration, database Database) error {
+func DownsampleOverTime(interval time.Duration, downsample_type string, database Database) error {
 	downsampleTicker := time.NewTicker(time.Duration(interval))
 
 	for range downsampleTicker.C {
-		err := database.Downsample(time.Now())
+		if downsample_type == "DOWNSAMPLE" {
```

Comment:
I think the different Downsample types should be constant strings in the database interface file, rather than magic strings